pkg/pgmodel/querier/query_builder.go

```diff
@@ -242,7 +242,29 @@ func canAttemptPushdown(metadata *promqlMetadata) bool {
 	path := metadata.path // PromQL AST.
 	queryHints := metadata.queryHints
 	selectHints := metadata.selectHints
-	return extension.ExtensionIsInstalled && queryHints != nil && !hasSubquery(path) && selectHints != nil
+
+	// We can't push down without hints
+	if queryHints == nil || selectHints == nil {
+		return false
+	}
+
+	// We can't do pushdowns without the extension
+	if !extension.ExtensionIsInstalled {
+		return false
+	}
+
+	// We can't handle subqueries in pushdowns
+	if hasSubquery(path) {
+		return false
+	}
```

Comment:
The correct Go style for things like this is to use a switch statement, would make the code more compact and easier to read.

Reply:
TIL, thanks!